### PR TITLE
feat: `XamlRoot` properties, `RootScale`, basic support for `ContentIsland`

### LINKED ISSUE #16042: Support for `XamlRoot` properties and `Changed` event
### What would you like to be added

Currently the properties are not working reliably.

### Why is this needed

Should work reliably

### For which platform

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #16042

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What's left to-do

- [x] GTK window content on WSL seems to be sized incorrectly
- [x] Adjust for WPF Uno islands
- [x] Adjust X11
- [x] Adjust macOS
- [x] Adjust non-Skia targets
- [x] Validate `CoreWindow`-based windows

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs [hunk 1]
```diff
@@ -48,34 +48,40 @@ private void OnLoaded(object sender, RoutedEventArgs e)
...
-			}
+			// TODO MZ: Adjust
+			//if (XamlRoot.Bounds != _initialBounds)
+			//{
+			//	XamlRoot.Changed -= OnXamlRootChanged;
+			//}
+			//else
+			//{
+			//	Assert.Fail("The XamlRoot's bounds should not have changed after Loaded.");
+			//}
```
Comment (Youssef1313) [inline src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs:59]: Needs to be done in this PR?

Comment (ramezgerges) [reply src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs:59]: I'm not sure what exactly need this fix, but I remember it was blowing up the X11 target, so probably yes.

**Review by Youssef1313 — APPROVED** [review]
Not very carefully reviewed though :)

File: src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs:59]: I need to validate if that condition makes sense against WinUI, it is very odd

**Review by ramezgerges — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

File: src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs (hunk 1 above)
Comment (agneszitte) [reply src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs:59]: Just curious, about those TODOs, should they be linked to GitHub issues so they are not forgotten?

Comment (MartinZikmund) [reply src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml.cs]: Validated and these assertions are actually not valid and are not in line with WinUI behavior - hence removing

**Pull request merged**